src/platform/plugins/shared/discover/public/application/main/components/layout/discover_layout.tsx

```diff
@@ -277,13 +277,19 @@ export function DiscoverLayout({ stateContainer }: DiscoverLayoutProps) {
     [stateContainer]
   );
 
-  const onFieldEdited = useCallback(
-    async ({ removedFieldName }: { removedFieldName?: string } = {}) => {
+  const onFieldEdited: (options: {
+    editedDataView: DataView;
+    removedFieldName?: string;
+  }) => Promise<void> = useCallback(
+    async (options) => {
+      const { editedDataView, removedFieldName } = options || {
+        editedDataView: dataView,
+      };
```

Comment:
```suggestion
      const { editedDataView, removedFieldName } = options;
```
Nit: isn't `options` always defined here?